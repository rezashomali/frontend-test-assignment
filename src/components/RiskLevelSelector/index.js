import React from "react";
import PropTypes from "prop-types";

const RiskLevelSelector = ({
  onChangeRiskLevel,
  minRiskLevel,
  maxRiskLevel,
}) => {
  var defultRiskl = 10;
  const options = [];
  for (let k = 1; k <= maxRiskLevel; ++k) {
    options.push(
      <option key={k} value={k}>
        {k}
      </option>
    );
  }

  return (
    <div>
      Risk level:
      <select onChange={onChangeRiskLevel} defaultValue={defultRiskl}>
        {options}
      </select>
    </div>
  );
};

RiskLevelSelector.defaultProps = {
  minRiskLevel: 3,
  maxRiskLevel: 25,
  onChangeRiskLevel: () => {},
};

RiskLevelSelector.propTypes = {
  minRiskLevel: PropTypes.number,
  maxRiskLevel: PropTypes.number,
  onChangeRiskLevel: PropTypes.func,
};

export default RiskLevelSelector;
