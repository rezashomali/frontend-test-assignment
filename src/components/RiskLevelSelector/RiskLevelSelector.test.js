import React from "react";
import renderer from "react-test-renderer";
import RiskLevelSelector from "./index";

test("RiskLevelSelector component snapshot ", () => {
  const component = renderer.create(
    <RiskLevelSelector riskLevel={0} onChangeRiskLevel={() => {}} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
