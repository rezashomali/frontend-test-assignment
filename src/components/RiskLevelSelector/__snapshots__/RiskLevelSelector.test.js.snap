// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RiskLevelSelector component snapshot  1`] = `
<div>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
      data-shrink={true}
    >
      Risk level
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        aria-hidden={true}
        autoFocus={false}
        className="MuiSelect-nativeInput"
        onAnimationStart={[Function]}
        onChange={[Function]}
        required={false}
        tabIndex={-1}
        value={0}
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
        >
          <span>
            Risk level
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
